src/ucp/core/ucp_worker.c

```diff
@@ -2058,6 +2057,11 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
                                            NULL);
     worker->am.alignment = UCP_PARAM_VALUE(WORKER, params, am_alignment,
                                            AM_ALIGNMENT, 1);
+    worker->uuid         = UCP_PARAM_VALUE(WORKER, params, worker_id, WORKER_ID,
+                                           ucs_generate_uuid((uintptr_t)worker));
+    if (params->field_mask & UCP_WORKER_PARAM_FIELD_WORKER_ID) {
+        worker->flags |= UCP_WORKER_FLAG_WORKER_ID;
```

Comment:
remove UCP_WORKER_FLAG_WORKER_ID, and instead set ep create flags to send worker_id when ep flag is set - new flag like UCP_EP_FLAG_SOCKADDR_SEND_WORKER_ID